gem/gem.py

```diff
@@ -161,7 +161,7 @@ def __new__(cls, array):
             return super(Literal, cls).__new__(cls)
 
     def __init__(self, array):
-        self.array = asarray(array, dtype=float)
+        self.array = asarray(array, dtype=float) if asarray(array).dtype is numpy.dtype('int') else asarray(array)
```

Comment:
What is the reason for this?

Reply:
It looks a bit odd, but I think Nic is trying to cast integers to floats without casting complex to float.

Reply:
```
try:
   array = asarray(array, dtype=float)
except TypeError:
   array = asarray(array, dtype=complex)
```